### PR TITLE
Initial packaging for catch2.

### PR DESCRIPTION
**Pull request by mjscosta**
@conda-forge/help-c-cpp

Initial packaging for catch2.

### CODE REVIEW

**Comment by mjscosta** [comment]
Hi there's been a week since this pull request, can someone merge it? 
@conda-forge/help-c-cpp
Thanks

File: recipes/catch2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tgz
+  url: https://github.com/catchorg/Catch2/archive/v{{ version }}.tar.gz
+  sha256: 4c94a685557328eb1b0ed1017ca37c3a378742dc03b558cf02267b6ba8579577
+
+build:
+  skip: true  # [vc<14]
+  number: 1001
```
Comment (chrisburr) [inline recipes/catch2/meta.yaml]: The build number should start from `0` now the compiler migration is complete.

**Comment by chrisburr** [comment]
Sorry for the delay reviewing this, LGTM 👍

**Comment by mjscosta** [comment]
No problem, thanks!

**Pull request merged**